src/main/java/org/opensearch/ad/task/ADTaskCacheManager.java

```diff
@@ -767,4 +771,54 @@ public void clearPendingEntities(String detectorId) {
             hcTaskCaches.get(detectorId).clearPendingEntities();
         }
     }
+
+    /**
+     * Check if realtime task field value changed or not by comparing with cache.
+     * 1. If new field value is null, will consider this field as not changed.
+     * 2. If any field value changed, will consider the realtime task changed.
+     * 3. If realtime task cache not found, will consider the realtime task changed
+     *    and put new realtime task into cache.
+     * @param detectorId detector id
+     * @param newState new task state
+     * @param newInitProgress new init progress
+     * @param newError new error
+     * @return true if realtime task changed comparing with realtime task cache.
+     */
+    public boolean realtimeTaskChanged(String detectorId, String newState, Float newInitProgress, String newError) {
```

Comment:
Seems to me that the name of this method would imply that it is just returning a boolean on whether the RT task has changed or not, and not also potentially updating the existing cache. Thoughts on changing the name to something more clear?

Reply:
Sure, how about `checkIfRealtimeTaskChangedAndUpdateCache`?

Reply:
I'm ok with that, sure